addon/jsonapi-attribute-transformer.js

```diff
@@ -71,20 +72,32 @@ var JSONAPIAttributeTransformer = function (/*store*/) {
    Recursively descend into the fixture json, looking for relationships
    whose attributes need transforming
    */
-  var findRelationships = function (fixture) {
+  var findRelationships = function (modelName, fixture) {
     var relationships = fixture.relationships;
     for (var key in relationships) {
       var data = relationships[key].data;
       if (Ember.typeOf(data) === 'array') {
         for (var i = 0, len = data.length; i < len; i++) {
-          transformAttributes(data[i]);
+          transformAttributes(modelName, data[i]);
         }
       } else {
-        transformAttributes(data);
+        transformAttributes(modelName, data);
       }
     }
-    transormObjectKeys(fixture.relationships);
+    transformRelationshipObjectKeys(modelName, fixture.relationships);
   };
+
+  var transformRelationshipObjectKeys = function(modelName, object) {
+    var serializer = store.serializerFor(modelName);
+    var transformFunction = serializer.keyForRelationship || Ember.String.dasherize;
+    for (var key in object) {
+      var value = object[key];
+      var newKey = transformFunction(key);
+      delete object[key];
+      object[newKey] = value;
+    }
+  };
```

Comment:
This is pretty much a duplicate of `transformObjectKeys`, except it uses `serializer.keyForRelationship`.